runtime/doc/api.txt

```diff
@@ -3108,6 +3108,195 @@ nvim_tabpage_set_var({tabpage}, {name}, {value})
+                            • name: (string) The name of the augroup
+                            • clear: (bool), defaults to true. Whether to
+                              clear existing command or not for the group (if
+                              it exists). See |autocmd-groups|.
+
+                Return: ~
+                    Integer The id of the created augroup.
+
+                See also: ~
+                    autocmd-groups
+
+nvim_create_autocmd({*opts})                           *nvim_create_autocmd()*
+                Define a new autocommand, See |autocmd|.
+
+                Example: >
+                    vim.api.nvim_create_autocmd({
+                      event = {"BufEnter", "BufWinEnter"},
+                      pattern = {"*.c", "*.h"},
+                      callback = function() print("This buffer enters") end, 
+                      once = false,
+                    })
+<
+
+                Example values for patterns: >
+                  pattern = "*.py,*.pyi"
+                  pattern = "*.py"
+                  pattern = {"*.py"}
+                  pattern = { "*.py", "*.pyi" }
+                  -- NOT supported
+                  pattern = {"*.py,*.pyi"}
+<
+
+                Examples values for events: >
+                  event = "FileType,CursorHold"
+                  event = "BufPreWrite"
+                  event = {"BufPostWrite"}
+                  event = {"CursorHold", "BufPreWrite", "BufPostWrite"}
+<
+
+                NOTE: Requires one of pattern or buffer, requires one of
+                callback or command,
+
+                Parameters: ~
+                    {opts}  Dictionary of Autocommand options.
+                            • group: (string) optional, the group name
+                              |autocmd-groups|.
+                            • event: (string|array) required, event or events
+                              to match against. See |autocmd-events|.
+                            • pattern: (string|array) optional, pattern or
+                              patterns to match against. See
+                              |autocmd-pattern|.
+                            • buffer: (integer) optional, buffer number for
+                              buffer local autocmds (can't be used together
+                              with pattern). See |autocmd-buflocal|.
+                            • desc: (string) optional, description of the
+                              autocmd.
+                            • callback: (function|string) optional, Lua
+                              function or Vim function (as string) to execute
+                              on event.
+                            • command (string) optional, vimscript command to
+                              execute on event (can't be used together with
+                              callback). Eg. opts.command = "let g:value_set =
+                              v:true"
```

Comment:
I think the example can be omitted here; I'd rather have a further examples block for `command` and `callback` at the top.

Reply:
Especially since the presence of both `command` and `callback` may lead to confusion, so a clear explanation of the distinction would be very helpful.

Reply:
Just added a better example at the start.

Reply:
Yes, that's very good!